h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -827,4 +829,80 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  /**
+   * Calculates means of given numerical Vectors. Provided there is a NaN on a row in any of the give vectors,
```

Comment:
What differentiates this method from say Vec.mean()?

Reply:
`Vec.mean` uses rollup stats.

```java
  /** Vecs's mean 
   *  @return Vec's mean */
  public double mean() { 
    return rollupStats()._mean; }
```

Those are not calculated for categorical columns. I'd have to duplicate the categorical vec and make it `T_NUM`, as we're unable to modify the original frame. Calculating it manually prevents us from copying the categorical data.

Also, if we'll ever implement the better version of Pearson (Perason can be implemented 100% without data copying in a very fast way), it will be reused there as well.

Reply:
it seem the way this method works is very specific to your implementation (if there is NA in any item of a given row, the whole row is skipped) - should this be move closer to your implementation?

Reply:
Yes, why not. Consider it done. I'll wait for the tests before pushing again, as I tested it on a cluster of 5 locally, but want the latest results.

Reply:
Moved it to AstSpearman. If it will be reused elsewhere, I'll make the necessary modifications afterwards.